search/django/weedid/utils.py

```diff
@@ -137,6 +137,17 @@ def remove_entity_local_record(user_id, upload_id):
             os.remove(dir_path)
 
 
+def atomic_copy(repository_dir, download_dir, tmp_dir, upload_id):
+    repo_path = tmp_dir / upload_id
+    zip_path = tmp_dir / f"{upload_id}.zip"
+    if os.path.isdir(repo_path):
+        copytree(repo_path, repository_dir / upload_id)
```

Comment:
Is there a reason not to use shutil.move? It should be much faster and atomic. copytree is not robust to I/O failure. I think our previous troubles with shutil.move were due to selinux